### PR TITLE
executors: add docs for executors on compose and k8s

### PR DESCRIPTION
**Pull request by davejrt**
Adds two deployment guides for those wishing to deploy executors on kubernetes (via helm or manifests) or docker-compose

## Test plan

Green linters in CI

## App preview:

- [Web](https://sg-web-dt-executor-docs.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: doc/admin/deploy_executors_docker.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Deploying Sourcegraph executors using Docker Compose
+
+<aside class="beta">
+<p>
+<span class="badge badge-beta">Beta</span> This feature is in beta and might change in the future.
```
Comment (Piszmog) [inline doc/admin/deploy_executors_docker.md:5]: Let's update these with _Experimental_.

**Review by Piszmog — APPROVED** [review]

File: doc/admin/deploy_executors_kubernetes.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+5. Confirm executors are working are working by checking the _Executors_ page under _Site Admin_ > _Maintenance_.
+
+#### Deployment via Helm
+
+1. Clone the [deploy-sourcegraph-helm](https://github.com/sourcegraph/deploy-sourcegraph-helm) repository to your local machine.
+2. Run `cd deploy-sourcegraph-helm/charts/sourcegraph-executor`.
+3. Edit the `values.yaml` with any other customizations you may require.
+4. Run the following command:
+   1. `helm upgrade --install -f values.yaml --version 4.4.2 sg-migrator sourcegraph/sourcegraph-migrator`
+5. Confirm executors are working are working by checking the _Executors_ page under _Site Admin_ > _Maintenance_.
```
Comment (danieldides) [inline doc/admin/deploy_executors_kubernetes.md:39]: Executors are their own top-level section now. I think this is incorrect in a few places in the docs.

I believe the correct path would be:
> Site Admin > Executors > Instances

Screenshot 2023-02-17 at 11 39 29

File: doc/admin/deploy_executors_kubernetes.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+
+For more information on the components being deployed see the [Executors readme](https://github.com/sourcegraph/deploy-sourcegraph/blob/master/configure/executors/README.md).
+
+## Note
+
+Executors deployed via docker-compose do not use [Firecracker](executors.md#how-it-works), meaning they require [privileged access](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) to the docker daemon running in a sidecar alongside the executor pod.
```
Comment (danieldides) [inline doc/admin/deploy_executors_kubernetes.md:46]:
```suggestion
Executors deployed in kubernetes do not use [Firecracker](executors.md#how-it-works), meaning they require [privileged access](https://kubernetes.io/docs/tasks/configure-pod-container/security-context/) to the docker daemon running in a sidecar alongside the executor pod.
```

File: doc/admin/deploy_executors_docker.md [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+A [docker-compose file](https://github.com/sourcegraph/deploy-sourcegraph-docker/blob/master/docker-compose/executors/executor.docker-compose.yaml) is provided to deploy executors standlone, or alongside your existing Sourcegraph deployment.
+
+## Deployment
+
+### Prerequisites
+
+  - Install [Docker Compose](https://docs.docker.com/compose/) on the server
+  - Minimum Docker [v20.10.0](https://docs.docker.com/engine/release-notes/#20100) and Docker Compose [v1.29.0](https://docs.docker.com/compose/release-notes/#1290)
+  - Docker Swarm mode is **not** supported
+
```
Comment (danieldides) [inline doc/admin/deploy_executors_docker.md:20]: I'm not sure we need a step-by-step here but maybe a note saying to clone the [repo](https://github.com/sourcegraph/deploy-sourcegraph-docker/blob/master/docker-compose/executors/executor.docker-compose.yaml) and update the environment variables?

File: doc/admin/deploy_executors_docker.md [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+## Note
+
+Executors deployed via docker-compose do not use [Firecracker](executors.md#how-it-works), meaning they require [privileged access](https://docs.docker.com/engine/reference/run/#runtime-privilege-and-linux-capabilities) to the docker daemon running on the host.
+
+If you have security concerns, consider deploying via [terraform](deploy_executors_terraform.md) or [installing the binary](deploy_executors_binary.md) directly.
+
+
+
+
```
Comment (danieldides) [inline doc/admin/deploy_executors_docker.md:29]:
```suggestion
```

File: doc/admin/deploy_executors_kubernetes.md (hunk 2 above)
Comment (davejrt) [reply doc/admin/deploy_executors_kubernetes.md:39]: TY!

**Pull request merged**